tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -388,4 +388,79 @@ var _ = Describe("odo devfile push command tests", func() {
 		})
 	})
 
+	Context("push with listing the devfile component", func() {
+
+		It("checks components in a specific app and all apps", func() {
+
+			// component created in "app" application
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			output := helper.CmdShouldPass("odo", "push", "--context", context)
+			Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+
+			// component created in different application
+			context2 := helper.CreateNewContext()
```

Comment:
we should delete the context before closing the test

Reply:
done